nvflare/app_opt/tracking/mlflow/mlflow_writer.py

```diff
@@ -79,21 +79,31 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
 
         Args:
             metrics (dict): Dictionary of metric_name: String -> value: Float. Note that some special values such as +/-
-                       Infinity may be replaced by other values depending on the store. For example, sql based store
-                       may replace +/- Infinity with max / min float values.
+                Infinity may be replaced by other values depending on the store. For example, sql based store
+                may replace +/- Infinity with max / min float values.
             step (int, optional): A single integer step at which to log the specified Metrics. If unspecified, each metric is
-                    logged at step zero.
+                logged at step zero.
         """
         self.sender.add(tag="metrics", value=metrics, data_type=AnalyticsDataType.METRICS, global_step=step)
 
+    def log_text(self, text: str, artifact_file_path: str) -> None:
+        """Log text as an artifact under the current run.
+
+        Args:
+            text (str): String of text to log.
+            artifact_file_path (str): The run-relative artifact file path in posixpath format
+                to which the text is saved (e.g. “dir/file.txt”).
+        """
+        self.sender.add(tag="text", value=text, data_type=AnalyticsDataType.TEXT, path=artifact_file_path)
```

Comment:
Can you help add these as test cases for the "sender add method"?

(log_text, set_tag, log_metrics) just make sure they all work.

Reply:
Looked into this, and I think that to do this right, there may need to be additional infrastructure around mocking the engine to test for events firing as expected. The scope of this is a bit larger than I expected at first, so maybe we can do this in a future PR.